[Ubuntu 25.04 VM] Fix: Broken Image URL in Setup (404 Error)

### ✅ Have you read and understood the above guidelines?

yes

### 🔎 Did you run the script with verbose mode enabled?

Yes, verbose mode was enabled and the output is included below

### 📜 What is the name of the script you are using?

ubuntu2504-vm.sh

### 📂 What was the exact command used to execute the script?

bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/vm/ubuntu2504-vm.sh)"

### ⚙️ What settings are you using?

- [x] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 13

### 📈 Which Proxmox version are you on?

pve-manager/9.1.4/5ac30304265fbd8e (running kernel: 6.17.4-2-pve)

### 📝 Provide a clear and concise description of the issue.

The script [ubuntu2504-vm.sh](https://github.com/community-scripts/ProxmoxVE/blob/main/vm/ubuntu2504-vm.sh) fails at [line 500](https://github.com/community-scripts/ProxmoxVE/blob/main/vm/ubuntu2504-vm.sh#L500) because the download URL for the Ubuntu 25.04 (Plucky) cloud image returns a 404 error. The current path in the script is no longer active as the builds have been moved to the reference/release archives.

https://github.com/community-scripts/ProxmoxVE/blob/c8dc0cffe0f9914640df8d1372b4827bbccaacc1/vm/ubuntu2504-vm.sh#L497

https://github.com/community-scripts/ProxmoxVE/blob/c8dc0cffe0f9914640df8d1372b4827bbccaacc1/vm/ubuntu2504-vm.sh#L500

### 🔄 Steps to reproduce the issue.

1.  Run this command on your PVE Shell:
      `bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/vm/ubuntu2504-vm.sh)"`

2. Set your Settings & Start install of Ubuntu 25.04

3. After a couple of seconds you see an Error

Error Output:
> curl: (22) The requested URL returned error: 404
> [ERROR] in line 500: exit code 22: while executing command curl -f#SL -o "$(basename "$URL")" "$URL"



### ❌ Paste the full error output (if available).

  🆔  Virtual Machine ID: 124
  📦  Machine Type: i440fx
  💾  Disk Size: 7G
  💾  Disk Cache: None
  🏠  Hostname: ubuntu
  🖥️  CPU Model: KVM64
  🧠  CPU Cores: 2
  🛠️  RAM Size: 2048
  🌉  Bridge: vmbr0
  🔗  MAC Address: 02:78:48:5F:60:A4
  🏷️  VLAN: Default
  ⚙️  Interface MTU Size: Default
  🌐  Start VM when completed: yes
  🚀  Creating a Ubuntu 25.04 VM using the above default settings
  ✔️  Using local-lvm for Storage Location.
  ✔️  Virtual Machine ID is 124.
  ✔️  https://cloud-images.ubuntu.com/plucky/current/plucky-server-cloudimg-amd64.img
curl: (22) The requested URL returned error: 404


[ERROR] in line 500: exit code 22: while executing command curl -f#SL -o "$(basename "$URL")" "$URL"

### 🖼️ Additional context (optional).

### How to: Change Links to fix

**Current Link [line 497](https://github.com/community-scripts/ProxmoxVE/blob/main/vm/ubuntu2504-vm.sh#L497)** : [...ubuntu.com/plucky/current/plucky-server-cloudimg-amd64.img](https://cloud-images.ubuntu.com/plucky/current/plucky-server-cloudimg-amd64.img)
https://github.com/community-scripts/ProxmoxVE/blob/c8dc0cffe0f9914640df8d1372b4827bbccaacc1/vm/ubuntu2504-vm.sh#L497

**Corrected Link** : [...ubuntu.com/releases/server/plucky/release/ubuntu-25.04-server-cloudimg-amd64.img](https://cloud-images.ubuntu.com/releases/server/plucky/release/ubuntu-25.04-server-cloudimg-amd64.img)

`URL=https://cloud-images.ubuntu.com/releases/server/plucky/release/ubuntu-25.04-server-cloudimg-amd64.img`

**Fixed File:**
[ubuntu2504-vm.sh](https://github.com/user-attachments/files/27463094/ubuntu2504-vm.sh)

--- a/vm/ubuntu2504-vm.sh
+++ b/vm/ubuntu2504-vm.sh
@@ -494,7 +494,7 @@ fi
 msg_ok "Using ${CL}${BL}$STORAGE${CL} ${GN}for Storage Location."
 msg_ok "Virtual Machine ID is ${CL}${BL}$VMID${CL}."
 msg_info "Retrieving the URL for the Ubuntu 25.04 Disk Image"
-URL=https://cloud-images.ubuntu.com/plucky/current/plucky-server-cloudimg-amd64.img
+URL=https://cloud-images.ubuntu.com/releases/server/plucky/release/ubuntu-25.04-server-cloudimg-amd64.img
 sleep 2
 msg_ok "${CL}${BL}${URL}${CL}"
 curl -f#SL -o "$(basename "$URL")" "$URL"

